### PR TITLE
Feature kafka consumer

### PR DESCRIPTION
**Pull request by SrishtiR**
Added consumer service to obtain json object from kafka topic containing sentiment score.

### CODE REVIEW

File: service/KafkaConsumerService.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      fromOffset: 'latest'
+    })
+  ]
+);
+
+const consumeTrendsFromKafka = () => {
+  consumer.on('message', message => {
+    const trends = JSON.parse(message.value);
+    logger.info('Received trends from Kafka topic');
+    insertAllTrendsLocationWise(trends);
```
Comment (shreshtha-pankaj) [inline service/KafkaConsumerService.js]: this will fail. Unless Redis cache service is updated.

File: service/KafkaConsumerService.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { insertAllTrendsLocationWise } from './RedisCacheService';
```
Comment (shreshtha-pankaj) [inline service/KafkaConsumerService.js:1]: this will fail. Unless Redis cache service is updated.

File: service/KafkaConsumerService.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import kafka from 'kafka-node';
+import { builtinModules } from 'module';
+import { logger } from './LoggerService';
+import util from 'util';
+
+let options = {
+  fromOffset: 'latest'
+};
+let kafkaHost = '35.222.250.101:9092'
```
Comment (shreshtha-pankaj) [inline service/KafkaConsumerService.js]: Lets put this in the config files we have.

File: service/KafkaConsumerService.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      autoCommit: false
+    },
+    (options = {
+      fromOffset: 'latest'
+    })
+  ]
+);
+
+const consumeTrendsFromKafka = () => {
```
Comment (shreshtha-pankaj) [inline service/KafkaConsumerService.js:26]: could you add on error as well?

File: service/KafkaConsumerService.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { insertAllTrendsLocationWise } from './RedisCacheService';
+import kafka from 'kafka-node';
+import { builtinModules } from 'module';
+import { logger } from './LoggerService';
+import util from 'util';
+
+let options = {
```
Comment (shreshtha-pankaj) [inline service/KafkaConsumerService.js:6]: put this in the configuration as well??

File: service/KafkaConsumerService.js (hunk 3 above)
Comment (AbigailFernandes) [inline service/KafkaConsumerService.js]: Should be const kafkaHost. The IP should come from Config.

**Review by shreshtha-pankaj — COMMENTED** [review]
I'll let you know how we can add things in the encrypted configuration today after or before class today.

File: service/KafkaConsumerService.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { insertAllTrendsLocationWise } from './RedisCacheService';
+import kafka from 'kafka-node';
+import { builtinModules } from 'module';
```
Comment (AbigailFernandes) [inline service/KafkaConsumerService.js]: This import seems redundant. You don't seem to be using it anywhere

File: service/KafkaConsumerService.js (hunk 5 above)
Comment (AbigailFernandes) [inline service/KafkaConsumerService.js:6]: You can remove this and always set it in the consumer object. This seems redundant too

File: service/KafkaConsumerService.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  fromOffset: 'latest'
+};
+let kafkaHost = '35.222.250.101:9092'
+const consumerClient = new kafka.KafkaClient({kafkaHost : kafkaHost});
+const kafkaTopic = 'trendSentiment'
+
+const consumer = new kafka.Consumer(
+  consumerClient,
+  [{ topic: kafkaTopic, partition: 0, fromOffset: -1 }],
+  [
```
Comment (AbigailFernandes) [inline service/KafkaConsumerService.js:16]: This should instead be 
```
  [
    {
      autoCommit: false
    },
  {
      fromOffset: 'latest'
    }
  ]
```

**Review by AbigailFernandes — CHANGES_REQUESTED** [review]

**Review by juanhotencoding — CHANGES_REQUESTED** [review]
have the changes been made?

**Review by shreshtha-pankaj — APPROVED** [review]

**Pull request merged**